packages/core/saved-objects/core-saved-objects-migration-server-internal/src/document_migrator/model_version.test.ts

```diff
@@ -123,65 +126,135 @@ describe('convertModelVersionTransformFn', () => {
-  });
+      const transform = convertModelVersionTransformFn({
+        log,
+        modelVersion: 1,
+        virtualVersion: '10.1.0',
+        definition,
+        type: 'up',
+      });
 
-  it('returns the document from the model transform', () => {
-    const upTransform = createModelTransformFn();
+      const doc = createDoc();
 
-    const resultDoc = createDoc();
-    upTransform.mockImplementation((doc) => {
-      return { document: resultDoc };
+      const result = transform(doc);
+      expect(result).toEqual({
+        transformedDoc: resultDoc,
+        additionalDocs: [],
+      });
     });
+  });
 
-    const definition: SavedObjectsModelVersion = {
-      modelChange: {
-        type: 'expansion',
-        transformation: { up: upTransform, down: jest.fn() },
-      },
-    };
+  describe('down transformation', () => {
+    it('generates a transform function calling the model transform', () => {
+      const upTransform = createModelTransformFn();
+      const downTransform = createModelTransformFn();
 
-    const transform = convertModelVersionTransformFn({
-      log,
-      modelVersion: 1,
-      virtualVersion: '10.1.0',
-      definition,
+      const definition: SavedObjectsModelVersion = {
+        modelChange: {
+          type: 'expansion',
+          transformation: { up: upTransform, down: downTransform },
+        },
+      };
+
+      const transform = convertModelVersionTransformFn({
+        log,
+        modelVersion: 1,
+        virtualVersion: '10.1.0',
+        definition,
+        type: 'down',
+      });
+
+      expect(upTransform).not.toHaveBeenCalled();
+      expect(downTransform).not.toHaveBeenCalled();
+
+      const doc = createDoc();
+      const context = { log, modelVersion: 1 };
+
+      transform(doc);
+
+      expect(upTransform).not.toHaveBeenCalled();
+      expect(downTransform).toHaveBeenCalledTimes(1);
+      expect(downTransform).toHaveBeenCalledWith(doc, context);
     });
 
-    const doc = createDoc();
+    it('returns the document from the model transform', () => {
+      const downTransform = createModelTransformFn();
+
+      const resultDoc = createDoc();
+      downTransform.mockImplementation((doc) => {
+        return { document: resultDoc };
+      });
+
+      const definition: SavedObjectsModelVersion = {
+        modelChange: {
+          type: 'expansion',
+          transformation: { up: jest.fn(), down: downTransform },
+        },
+      };
+
+      const transform = convertModelVersionTransformFn({
+        log,
+        modelVersion: 1,
+        virtualVersion: '10.1.0',
+        definition,
+        type: 'down',
+      });
+
+      const doc = createDoc();
 
-    const result = transform(doc);
-    expect(result).toEqual({
-      transformedDoc: resultDoc,
-      additionalDocs: [],
+      const result = transform(doc);
+      expect(result).toEqual({
+        transformedDoc: resultDoc,
+        additionalDocs: [],
+      });
```

Comment:
nit: should we check that `downTransform` has been called once?

Reply:
it's covered in the previous test